### DISCUSSION TITLE
#ASK How to access cookies from a dynamic variable in a REST API and use them for other API requests

### DISCUSSION THREAD

**Discussion by memcaliber** [Budibase/budibase#12265] [Help]
**Checklist**
- [x] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**
- Self
  - Method:  docker single image
  - Budibase Version: 2.11.45.

**Describe the bug**
I think this is not a bug, but it is more appropriately termed an issue. Since I am still a beginner, I do not know how to create an issue to ask about it. Please understand.

I would like to ask about REST Dynamic Variables. I have followed the explanation in https://docs.budibase.com/docs/rest-variables.

I have accessed the endpoint http://192.168.100.53:5000/api/login and followed the steps. The Dynamic Variable, which is an access-token, has been successfully stored. I have received the value in cookies sent by the server through the response header.

Then, I created a new data source and continued to use the endpoint http://192.168.100.53:5000/api/users. In the Headers, I added a header:

Key: Authorization
Value: Bearer {{ Dynamic.cookie }}

**Expected behavior**
My expectation was to access the cookie I obtained when I successfully logged in. However, it turns out that the cookie I received earlier cannot be accessed, and as a result, my request to http://192.168.100.53:5000/api/users is rejected because no access-token is sent.
My question is, how can I make the cookie from the Login (http://192.168.100.53:5000/api/login) accessible or recognized by other endpoints?

**Screenshots**
1. Screenshots 1 : About Dynamic Variables after login successfully. (Endpoint 1)
2. Screenshots 2 : About Setting on Data Source with Endpoint 2
3. Screenshots 3 : Response from server because no access-token is sent
![budi-01](https://github.com/Budibase/budibase/assets/17555144/517ad854-44b8-4982-8286-96127049d41c)
![budi-02](https://github.com/Budibase/budibase/assets/17555144/9d609f79-c4d5-4c10-bd80-8fdf2cc46c44)
![budi-03](https://github.com/Budibase/budibase/assets/17555144/4e38936c-f8fc-4c32-874c-dc1d108c7a06)

With all due respect, I thank you for your response.

**Comment by mike12345567** [comment]
Hi @memcaliber - I've converted this to a discussion as this is a better place for getting help with issues rather than bugs.
